### PR TITLE
Move OC.AppConfig, OC.EventSource and OC.L10N to the server bundle

### PR DESCRIPTION
**Pull request by ChristophWurst**
And another three requests less per page load :rocket:

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]

File: core/src/OC/index.js [hunk 1]
```diff
@@ -34,6 +37,9 @@ Object.assign(Backbone, {
 /** @namespace OC */
 export default {
 	Apps,
+	AppConfig,
 	Backbone,
 	ContactsMenu,
+	EventSource,
+	L10N,
```
Comment (skjnldsv) [inline core/src/OC/index.js]: god I'm loving this <3
It's so beautiful I could cry :eyes:

**Review by skjnldsv — APPROVED** [review]

File: core/src/OC/index.js (hunk 1 above)
Comment (ChristophWurst) [reply core/src/OC/index.js]: :heart: :dancer: :joy_cat:

**Comment by MorrisJobke** [comment]
> And another three requests less per page load 🚀

I was wondering why this is the case, but those files are only loaded when `debug` is set to `true`. Otherwise those files are already bundled 😉

**Comment by ChristophWurst** [comment]
> I was wondering why this is the case, but those files are only loaded when `debug` is set to `true`. Otherwise those files are already bundled wink

Psssst :wink:

**Comment by ChristophWurst** [comment]
But they relied on server-side bundling, hence always required the server to serve them when requested. With the webpack they are static and can be served by the webserver.

**Comment by rullzer** [comment]
> But they relied on server-side bundling, hence always required the server to serve them when requested. With the webpack they are static and can be served by the webserver.

Exactly. And this is something that should not be underestimated. The bundle here can be servered statically by any webserver. Proper caching gzip etc. While the other always needed to go to the storage when serving or at least setup the whole nextcloud php env to serve the file. And it is worse when a relatively slow stroage is used (objectstore requies http requests for example).

Long story short :rocket:

**Comment by MorrisJobke** [comment]
Sure - I didn't meant to made this worse than it is. Even if it just would mean that we properly organize our JS code it would be perfectly fine 👍

**Pull request merged**